libcudacxx/include/cuda/__memory_resource/resource_ref.h

```diff
@@ -425,54 +453,66 @@ class basic_resource_ref
   template <class...>
   friend struct _Resource_vtable;
 
+  friend struct _Resource_ref_helper;
+
   using __vtable = _Filtered_vtable<_Properties...>;
 
   template <class... _OtherProperties>
   static constexpr bool __properties_match =
-    _CUDA_VSTD::__all_of<_CUDA_VSTD::_One_of<_Properties, _OtherProperties...>...>;
+    _CUDA_VSTD::__type_set_contains<_CUDA_VSTD::__type_set<_OtherProperties...>, _Properties...>;
```

Comment:
I do not understand the need for this to be more generic than it currently is. 

We want to ensure that the properties match exactly. Having derived properties seems wrong

Reply:
this isn't testing for derived properties. this is testing for set membership. the `__all_of`/`_One_of` is an O(n^2) algorithm. with `__type_set` it is O(2n).